src/view/utils/segments.js

```diff
@@ -0,0 +1,107 @@
+import { getRawId } from './prompts';
+import { getCriteria } from '../../criteria/utils';
+
+const day = 1000 * 60 * 60 * 24;
+export const periods = {
+	day,
+	week: day * 7,
+	month: day * 30,
+};
+
+/**
+ * Whether the reader matches the segment criteria.
+ *
+ * @param {Object} segmentCriteria Segment criteria.
+ *
+ * @return {boolean} True if the reader matches all of the segment's criteria, false if not.
+ */
+export const match = segmentCriteria => {
+	for ( const item of segmentCriteria ) {
+		const criteria = getCriteria( item.criteria_id );
+		if ( ! criteria ) {
+			continue;
+		}
+		if ( ! criteria.matches( item ) ) {
+			return false;
+		}
+	}
+	return true;
+};
+
+/**
+ * Get the reader's highest-priority segment match.
+ *
+ * @param {Object} segments Segments.
+ *
+ * @return {string|null} Segment ID, or null.
+ */
+export const getBestPrioritySegment = segments => {
+	const matchingSegments = [];
+	for ( const segmentId in segments ) {
+		if ( match( segments[ segmentId ].criteria ) ) {
+			matchingSegments.push( {
+				id: segmentId,
+				priority: segments[ segmentId ].priority,
+			} );
+		}
+	}
+
+	if ( ! matchingSegments.length ) {
+		return null;
+	}
+
+	matchingSegments.sort( ( a, b ) => a.priority - b.priority );
+
+	return matchingSegments[ 0 ].id;
+};
+
+/**
+ * Check the reader's activity against a given prompt's assigned segments.
+ *
+ * @param {HTMLElement} prompt          HTML element of the prompt being checked.
+ * @param {string}      matchingSegment ID of the reader's highest-priority matching segment.
+ * @param {Object}      ras             Reader Data Library object.
+ * @return {boolean} True if the prompt should be displayed, false if not.
+ */
+export const shouldPromptBeDisplayed = ( prompt, matchingSegment, ras ) => {
```

Comment:
Answering your question on unit testing, I think this is a great candidate. It'll require mocking the `ras` object but otherwise, it's a core function that connects a lot of pieces.

Reply:
This entire file, actually 😄

Reply:
Tests added in 355972c